### ISSUE TITLE
HDF5 failed to load on Julia 0.4 dev

### ISSUE THREAD

**Issue by siyideng** [timholy/HDF5.jl#231]
```
julia> using HDF5
ERROR: LoadError: LoadError: MethodError: `start` has no method matching start(::Type{HDF5
.DimSize{N}})
 in append_any at base.jl:198
 in include at boot.jl:250
 in include_from_node1 at loading.jl:129
 in include at boot.jl:250
 in include_from_node1 at loading.jl:129
 in reload_path at loading.jl:153
 in _require at loading.jl:68
 in require at loading.jl:51
while loading C:\Users\sdeng\.julia\v0.4\HDF5\src\plain.jl, in expression starting on line
 437
while loading C:\Users\sdeng\.julia\v0.4\HDF5\src\HDF5.jl, in expression starting on line
1
```

**Comment by timholy** [comment]
This will be fixed by #228, once merged.

**Comment by simonster** [comment]
Fixed by #232, the successor to #228.
